src/Compilers/Core/Portable/Syntax/SeparatedSyntaxList.cs

```diff
@@ -592,9 +592,16 @@ IEnumerator IEnumerable.GetEnumerator()
             return new SeparatedSyntaxList<SyntaxNode>(nodes._list);
         }
 
+        [Obsolete("Implicit downcast is not safe. Use CastDown<TNode>() instead", error: true)]
         public static implicit operator SeparatedSyntaxList<TNode>(SeparatedSyntaxList<SyntaxNode> nodes)
         {
             return new SeparatedSyntaxList<TNode>(nodes._list);
         }
+
+        public SeparatedSyntaxList<TDerived> CastDown<TDerived>()
+            where TDerived : TNode
+        {
+            return new SeparatedSyntaxList<TDerived>(_list);
```

Comment:
@CyrusNajmabadi However, the *API* provides a safe alternative. Since this type is immutable, we have the option of casting each element of `_list` here to force an `InvalidCastException` as early as possible, but I have a hard time balancing that against the performance of things like `ISyntaxFactsService` which always return an upcast list.

Would renaming this method to `UncheckedCastDown` resolve your concern?